src/win/fs.c

```diff
@@ -148,6 +148,72 @@ void uv_fs_init(void) {
 }
 
 
+static int32_t fs__decode_wtf8_char(const char** input) {
+  uint32_t code_point;
+  uint8_t b, b2, b3, b4;
+
+  b = **input;
+  if (b <= 0x7F) {
+    code_point = b;
+  } else if (b >= 0xC2 && b <= 0xDF) {
+    if (!(*input)[1])
+      return -1;
+    b2 = *++*input;
+    code_point = ((b & 0x1F) << 6) + (b2 & 0x3F);
+  } else if (b >= 0xE0 && b <= 0xEF) {
+    if (!(*input)[1] || !(*input)[2])
+      return -1;
+    b2 = *++*input;
+    b3 = *++*input;
+    code_point = ((b & 0x0F) << 12) + ((b2 & 0x3F) << 6) + (b3 & 0x3F);
+  } else if (b >= 0xF0 && b <= 0xF0) {
+    if (!(*input)[1] || !(*input)[2] || !(*input)[3])
+      return -1;
+    b2 = *++*input;
+    b3 = *++*input;
+    b4 = *++*input;
+    code_point = ((b & 0x07) << 18) + ((b2 & 0x3F) << 12) +
+      ((b3 & 0x3F) << 6) + (b4 & 0x3F);
+  } else {
+    return -1;
+  }
+  if (code_point > 0x10ffff)
+    return -1;
+  return (int32_t)code_point;
+}
+
+
+static int fs__get_length_wtf8(const char* source_ptr) {
+  int w_target_len = 0;
+  do {
+    int32_t code_point;
+    code_point = fs__decode_wtf8_char(&source_ptr);
+    if (code_point < 0)
+      return -1;
+    if (code_point > 0xFFFF)
+      w_target_len++;
+    w_target_len++;
+  } while (*source_ptr++);
+  return w_target_len;
+}
+
+
+static void fs__wtf8_to_wide(const char* source_ptr, WCHAR* w_target) {
+  do {
+    int32_t code_point;
+    code_point = fs__decode_wtf8_char(&source_ptr);
+    // fs__get_length_wtf8 should have been called and checked first
```

Comment:
style nit: use C-style comments and full sentences:
```suggestion
    /* fs__get_length_wtf8 should have been called and checked first. */
```